src/app/components/chat/client.ts

```diff
@@ -193,7 +229,21 @@ async function connect(chat: ChatClient) {
 		setChatRoom(chat, undefined);
 	});
 
+	chat.socket.onError((e: any) => {
+		console.warn('[Chat] Got error from socket');
+		console.warn(e);
+		chat.reconnect();
+	});
+
+	chat.socket.onClose(() => {
+		console.warn('[Chat] Socket closed unexpectedly');
+		chat.reconnect();
+	});
+
+	chat.socket.sendHeartbeat();
```

Comment:
the @types seem to not have this function defined, but phoenix's js also doesn't have this marked as private, so I guess this is valid?

Reply:
Think we just need a quick comment since this is marked as an error in TS

Reply:
oh i don't think i should've commented the sendHeartbeat, i was debugging with it, is all.